nyaa/routes.py

```diff
@@ -696,8 +697,10 @@ def site_help():
 
 
 # #################################### API ROUTES ####################################
-# DISABLED FOR NOW
 @app.route('/api/upload', methods=['POST'])
 def api_upload():
-    api_response = api_handler.api_upload(flask.request)
+    is_valid_user, user, debug = api_handler.validate_user(flask.request)
+    if not is_valid_user:
+        return flask.make_response(flask.jsonify({"Failure": "Invalid username or password."}), 400)
```

Comment:
I'd really prefer `failure` instead of `Failure` - this is an API, for computers. Lowercase keys!